docs/draft/6-payloads.md

```diff
@@ -0,0 +1,359 @@
+      - [Payload](#payload-1)
+      - [Content types](#content-types)
+        - [Sticker content type](#sticker-content-type)
+        - [Image content type](#image-content-type)
+      - [Message types](#message-types)
+      - [Clock vs Timestamp and message ordering](#clock-vs-timestamp-and-message-ordering)
+      - [Chats](#chats)
+    - [Contact Update](#contact-update)
+      - [Payload](#payload-2)
+      - [Contact update](#contact-update-1)
+    - [SyncInstallationContact](#syncinstallationcontact)
+      - [Payload](#payload-3)
+    - [SyncInstallationPublicChat](#syncinstallationpublicchat)
+      - [Payload](#payload-4)
+    - [PairInstallation](#pairinstallation)
+      - [Payload](#payload-5)
+    - [MembershipUpdateMessage and MembershipUpdateEvent](#membershipupdatemessage-and-membershipupdateevent)
+  - [Upgradability](#upgradability)
+  - [Security Considerations](#security-considerations)
+  - [Design rationale](#design-rationale)
+
+## Introduction
+
+In this document we describe the payload format and some special considerations.
+
+## Payload wrapper
+
+All payloads are wrapped in a [protobuf record](https://developers.google.com/protocol-buffers/)
+record:
+
+```protobuf
+message StatusProtocolMessage {
+  bytes signature = 4001;
+  bytes payload = 4002;
+}
+```
+
+`signature` is the bytes of the signed `SHA3-256` of the payload, signed with the key of the author of the message.
+The signature is needed to validate authorship of the message, so that the message can be relayed to third parties.
+If a signature is not present but an author is provided by a layer below, the message is not to be relayed to third parties and it is considered plausibly deniable.
+
+## Encoding
+
+The payload is encoded using [Protobuf](https://developers.google.com/protocol-buffers)
+
+## Types of messages
+
+### Message
+
+The type `ChatMessage` represents a chat message exchanged between clients.
+
+#### Payload
+
+The protobuf description is:
+
+```protobuf
+message ChatMessage {
+  // Lamport timestamp of the chat message
+  uint64 clock = 1;
+  // Unix timestamps in milliseconds, currently not used as we use whisper as more reliable, but here
+  // so that we don't rely on it
+  uint64 timestamp = 2;
+  // Text of the message
+  string text = 3;
+  // Id of the message that we are replying to
+  string response_to = 4;
+  // Ens name of the sender
+  string ens_name = 5;
+  // Chat id, this field is symmetric for public-chats and private group chats,
+  // but asymmetric in case of one-to-ones, as the sender will use the chat-id
+  // of the received, while the receiver will use the chat-id of the sender.
+  // Probably should be the concatenation of sender-pk & receiver-pk in alphabetical order
+  string chat_id = 6;
+
+  // The type of message (public/one-to-one/private-group-chat)
+  MessageType message_type = 7;
+  // The type of the content of the message
+  ContentType content_type = 8;
+
+  oneof payload {
```

Comment:
Do we specify message size somewhere? Might be good to do it explicit in payload / image spec, e..g Status supports messages up to <X size>

Reply:
Currently it's a bit tricky, we can set a limit, but there's no guarantee that it will get through, as each whisper node sets its own max message limit. 
But we can suggest a safe limit, but we need to do a bit more exploratory work I think, to check image sizes , pow etc